src/cloud-api-adaptor/pkg/adaptor/server.go

```diff
@@ -80,7 +80,7 @@ func (s *server) Start(ctx context.Context) (err error) {
 	// Advertise node resources
 	err = k8sops.AdvertiseExtendedResources(s.PeerPodsLimitPerNode)
 	if err != nil {
-		return err
+		logger.Printf("failed AdvertiseExtendedResources: %s", err)
```

Comment:
Hi @spotlesstofu !

I'm afraid that logging a message instead of return an error it will be masking a fatal failure, when running with Kubernetes.

I looked at `AdvertiseExtendedResources()` implementation, in the line https://github.com/confidential-containers/cloud-api-adaptor/blob/5708832ad53c8450baa314bc71c059de4ba7d433/src/cloud-api-adaptor/pkg/adaptor/k8sops/node.go#L28, it gets the `NODE_NAME` which should be non-empty when running from k8s. What if you put a guard there, so that if `NODE_NAME` is empty then it returns `nil` (logging a message too)?

Reply:
Thanks @wainersm.

Building on your feedback, I introduced a function to gate the Kubernetes-specific feature.

What do you think?